components/camel-dynamic-router/src/main/java/org/apache/camel/component/dynamicrouter/DynamicRouterConfiguration.java

```diff
@@ -131,6 +129,103 @@ public class DynamicRouterConfiguration {
+     * {@link org.apache.camel.Exchange} and the caused exception will be thrown. Will also stop if processing the
+     * exchange failed (has a fault message), or an exception was thrown and handled by the error handler (such as using
+     * onException). In all situations, the multicast will stop further processing. This is the same behavior as in the
+     * pipeline that is used by the routing engine. The default behavior is to not stop, but to continue processing
+     * until the end.
+     */
+    @UriParam(label = "common", defaultValue = "false")
+    private boolean stopOnException;
+
+    /**
+     * Ignore the invalid endpoint exception when attempting to create a producer with an invalid endpoint.
+     */
+    @UriParam(label = "common", defaultValue = "false")
+    private boolean ignoreInvalidEndpoints;
+
+    /**
+     * If enabled, then Camel will process replies out-of-order (e.g., in the order they come back). If disabled, Camel
+     * will process replies in the same order as defined by the multicast.
+     */
+    @UriParam(label = "common", defaultValue = "false")
+    private boolean streaming;
+
+    /**
+     * Sets a total timeout specified in milliseconds, when using parallel processing. If the Multicast has not been
+     * able to send and process all replies within the given timeframe, then the timeout triggers and the Multicast
+     * breaks out and continues. Notice that, if you provide a TimeoutAwareAggregationStrategy, then the timeout method
+     * is invoked before breaking out. If the timeout is reached with running tasks still remaining, certain tasks (for
+     * which it is difficult for Camel to shut down in a graceful manner) may continue to run. So use this option with a
+     * bit of care.
+     */
+    @UriParam(label = "common", defaultValue = "-1")
+    private long timeout;
+
+    /**
+     * Uses the Processor when preparing the {@link org.apache.camel.Exchange} to be sent. This can be used to
+     * deep-clone messages that should be sent, or to provide any custom logic that is needed before the exchange is
+     * sent.
+     */
+    @UriParam(label = "common")
+    private String onPrepare;
+
+    /**
+     * Shares the {@link org.apache.camel.spi.UnitOfWork} with the parent and each of the sub messages. Multicast will,
+     * by default, not share a unit of work between the parent exchange and each multicasted exchange. This means each
+     * sub exchange has its own individual unit of work.
+     */
+    @UriParam(label = "common", defaultValue = "false")
+    private boolean shareUnitOfWork;
+
+    /**
+     * Refers to a custom Thread Pool to be used for parallel processing. Notice that, if you set this option, then
+     * parallel processing is automatically implied, and you do not have to enable that option in addition to this one.
+     */
+    @UriParam(label = "common")
+    private String executorService;
+
+    /**
+     * TODO: not a configuration setting
```

Comment:
TODO here

Reply:
Indeed.  It is now TODONE -- I removed that option.